qiskit/transpiler/preset_passmanagers/level3.py

```diff
@@ -257,20 +262,35 @@ def _opt_control(property_set):
         # optimization loop to correct for incorrect directions that might be
         # inserted by UnitarySynthesis which is direction aware but only via
         # the coupling map which with a target doesn't give a full picture
-        if target is not None:
+        if target is not None and optimization is not None:
             optimization.append(
                 _opt + unroll + _depth_check + _size_check + _direction, do_while=_opt_control
             )
-        else:
+        elif optimization is not None:
```

Comment:
```
if X and Y:
  <branch 1>
elif Y:
   <branch 2>
else:
    <branch 3>
```
is harder to read than
```
if Y:
   if X:
      <branch 1>
   else:
      <branch 2>
else:
    <branch 3>
```

Reply:
I personally don't really see a functional difference both seem equally legible to me, but I'll change the logic here. I was going to do this as part of https://github.com/Qiskit/qiskit-terra/pull/8305#discussion_r933150401 too.

Reply:
Done: https://github.com/Qiskit/qiskit-terra/pull/8305/commits/4190774c3b50a1a9b7832437dd4492f8ddea4bfc